partcad/src/partcad/__init__.py

```diff
@@ -84,17 +91,77 @@ def before_send(event, hint):
         debug=user_config.get_bool("sentry.debug"),
         shutdown_timeout=user_config.get_int("sentry.shutdown_timeout"),
-        enable_tracing=True,
-        attach_stacktrace=False,
+        enable_tracing=user_config.get_bool("sentry.enable_tracing"),
+        attach_stacktrace=user_config.get_bool("sentry.attach_stacktrace"),
         traces_sample_rate=user_config.get_float("sentry.traces_sample_rate"),
         integrations=[
             LoggingIntegration(
-                level=logging.ERROR,
+                level=logging.logging.getLevelName(
+                    user_config.get_string("sentry.integrations.LoggingIntegration.level").upper()
+                )
             )
         ],
         before_send=before_send,
     )
+
+    # # https://develop.sentry.dev/sdk/data-model/event-payloads/user/
+    # # For some reason, the {{auto}} tag is not working:
+    # # "null - Replaced because of SDK configuration"
+    # sentry_sdk.set_user({"ip_address": "{{auto}}"})
+
+    sentry_sdk.set_tag("utm.source", "PartCAD")
+    sentry_sdk.set_tag("utm.medium", "Sentry")
+    sentry_sdk.set_tag("utm.campaign", "Open Source")
+    sentry_sdk.set_tag("utm.content", "API")
+
+    sentry_sdk.set_tag("env.remote_containers", os.environ.get("REMOTE_CONTAINERS", "false").lower() == "true")
+
+    hostname = socket.getfqdn()
+    hostname_md5 = hashlib.md5(hostname.encode()).hexdigest()
+    sentry_sdk.set_tag("hostname.md5", hostname_md5)
+
+    username = os.getenv("USER")
+    username_md5 = hashlib.md5(username.encode()).hexdigest()
+    sentry_sdk.set_tag("username.md5", username_md5)
+
+    cache_dir = os.path.expanduser("~/.cache/partcad")
+    os.makedirs(cache_dir, exist_ok=True)
+    cache_file = os.path.join(cache_dir, "id")
+
+    # Check if the ID is already cached
+    if os.path.exists(cache_file):
+        with open(cache_file, "r") as f:
+            unique_uid = f.read().strip()
+    else:
+        # Generate a new unique ID and cache it
+        unique_uid = str(uuid.uuid4())
+        with open(cache_file, "w") as f:
+            f.write(unique_uid)
+
+    sentry_sdk.set_user({"id": unique_uid})
+
+    # https://docs.sentry.io/product/performance/metrics/#custom-performance-measurements
+    # https://docs.sentry.io/platforms/python/tracing/instrumentation/performance-metrics/
+    memory_rss = psutil.Process().memory_info().rss
+    cpu_user = psutil.Process().cpu_times().user
+
+    # TODO: Report internal metrics, such as following:
+
+    # INFO:  PartCAD version: 0.7.63
+    # INFO:  Internal data storage location: /home/vscode/.partcad
+    # INFO:  Total internal data storage size: 0.00MB
+    # INFO:  Git cache size: 0.00MB
+    # INFO:  Tar cache size: 0.00MB
+    # INFO:  Runtime environments size: 0.00MB
+    # INFO:  DONE: Status: this: 0.07s
+
+    sentry_sdk.set_measurement("memory.rss", memory_rss, "byte")
+    sentry_sdk.set_measurement("cpu.user", cpu_user, "second")
+
+    sentry_sdk.profiler.start_profiler()
```

Comment:
```
[sentry] DEBUG: [Profiling] Setting up continuous profiler in thread mode
...
[sentry] DEBUG: [Profiling] Discarding profile because profiler was not started.
DEBUG:sentry_sdk.errors:[Profiling] Discarding profile because profiler was not started.
```